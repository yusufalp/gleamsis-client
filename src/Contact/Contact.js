import React from 'react';

class Contact extends React.Component {
  render() {
    return (
      <div>
        <header>
          <h2>Contact us with your inquiries</h2>
        </header>
        <main>
          <form action="https://formspree.io/f/xaypjzbq" method="POST">
            <label>Your name:
              <input type="text" required />
            </label>
            <label>Your email:
              <input type="email" name="_replyto" required />
            </label>
            <label>Your message:
              <textarea name="message" cols="20" rows="8" required></textarea>
            </label>
            <button type="submit">Send</button>
          </form>
        </main>
      </div>
    )
  }
}

export default Contact;