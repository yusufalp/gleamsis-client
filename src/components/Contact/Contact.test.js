import React from 'react';
import ReactDOM from 'react-dom';
import Contact from './Contact';
import renderer from 'react-test-renderer';

describe('Contact component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Contact />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<Contact />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})