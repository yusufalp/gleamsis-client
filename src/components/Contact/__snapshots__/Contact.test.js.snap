// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact component renders the UI as expected 1`] = `
<div>
  <header>
    <h2>
      Contact us with your inquiries
    </h2>
  </header>
  <main>
    <form
      action="https://formspree.io/f/xaypjzbq"
      method="POST"
    >
      <label>
        <input
          placeholder="Name"
          required={true}
          type="text"
        />
      </label>
      <label>
        <input
          name="_replyto"
          placeholder="Email"
          required={true}
          type="email"
        />
      </label>
      <label>
        <textarea
          cols="20"
          name="message"
          placeholder="Message"
          required={true}
          rows="8"
        />
      </label>
      <button
        type="submit"
      >
        Send
      </button>
      <p
        className="required"
      >
        All fields are required
      </p>
    </form>
  </main>
</div>
`;
